### PR TITLE
ADRV9002 Support for RX1TX1, RX2TX2 board variants; Differentiate tests for LVDS and CMOS profiles

### PR DESCRIPTION
**Pull request by HRosete**
# Description

I added support for the RX1TX1 and RX2TX2 board variants of ADRV9002 by differentiating the two as "adrv9002-rx1tx1" and "adrv9002_rx2tx2" respectively. Integrated them as a list in the "hardware" variable in the test.

I added "lvds_test" and "cmos_test" markers on the test itself to filter which profile tests are needed based on the Linux file used. Additional support, marker and command-line argument initialization were added in common.py.

Added TX2 and RX2 support for "dma_tx" and "dma_rx" in the dma_tests module.

Added a test_map for the keyword variable "adrv9002" under test_map.py.

# How has this been tested?

Tested in a Jenkins pipeline for board variants under Zed and UltraScale (ZCU102) platforms.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: test/common.py [hunk 1]
```diff
@@ -65,6 +81,22 @@ def pytest_runtest_setup(item):
...
+        )
+
+    # Handle CMOS tests
+    cmos = item.config.getoption("--cmos")
+    marks = [mark.name for mark in item.iter_markers()]
+    if not cmos and "cmos_test" in marks:
+        pytest.skip(
+            "CMOS testing disabled. Use --cmos flag to enable"
+        )
+
```
Comment (tfcollins) [inline test/common.py:99]: Can you add a check to make sure CMOS+LVDS are not enabled together.

**Comment by tfcollins** [comment]
Generally looks good. Linter caught a few things: https://github.com/analogdevicesinc/pyadi-iio/pull/211/checks?check_run_id=3072844909
Might have to add a word to the codespell whitelist in the repo root.

File: test/common.py (hunk 1 above)
Comment (HRosete) [reply test/common.py:89]: I now added one Travis, it's in 6c598a1, though I haven't had the chance to test it in the pipeline with an updated command in the CLI on jenkins-shared-lib. I ought to manually add either the --cmos or the --lvds flag and run the pipeline to check if it does work, but I don't have any access on the repo. Can you please check my code if it works as we intend it?

**Comment by HRosete** [comment]
> Generally looks good. Linter caught a few things: https://github.com/analogdevicesinc/pyadi-iio/pull/211/checks?check_run_id=3072844909
> Might have to add a word to the codespell whitelist in the repo root.

I ran the pre-commit checks and committed the changes. I forgot to run it earlier before making this PR.

**Pull request merged**